File: app/src/main/java/com/android_cademy/starwarstinder/di/AppComponent.java
package com.android_cademy.starwarstinder.di;

import com.android_cademy.starwarstinder.View.MainActivity;
import com.android_cademy.starwarstinder.network.ProfileFetchService;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(MainActivity mainActivity);

    void inject(ProfileFetchService profileFetchService);
}

File: app/src/main/java/com/android_cademy/starwarstinder/di/AppComponent.kt
package com.android_cademy.starwarstinder.di

import com.android_cademy.starwarstinder.view.MainActivity
import com.android_academy.network.ProfileFetchService

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {

    fun inject(mainActivity: MainActivity)

    fun inject(profileFetchService: ProfileFetchService)
}
